Permissions.py

```diff
@@ -0,0 +1,197 @@
+# vim: sw=4 et sts=4 ts=4 :
+#############################################################################
+# Author        : Matthias Gerstner
+# Purpose       : reusable code for parsing permissions/chkstat profiles
+#############################################################################
+
+import os
+import copy
+
+
+class PermissionsEntry:
+
+    # source profile path
+    profile = None
+    # source profile line nr
+    linenr = None
+    # target path
+    path = None
+    owner = None
+    group = None
+    # mode as integer
+    mode = None
+    caps = []
+    # related paths from variable expansions
+    related_paths = []
+
+    def __init__(self, _profile, _line_nr):
+
+        self.profile = _profile
+        self.linenr = _line_nr
+
+    def __str__(self):
+
+        ret = "{}:{}: {path} {owner}:{group} {mode}".format(
+            self.profile,
+            self.linenr,
+            path=self.path,
+            owner=self.owner,
+            group=self.group,
+            mode=oct(self.mode)
+        )
+
+        for cap in self.caps:
+            ret += "\n+capability " + cap
+
+        for related in self.related_paths:
+            ret += "\nrelated to " + related
+
+        return ret
+
+
+class VariablesHandler:
+
+    def __init__(self, variables_conf_path):
+
+        self.m_variables = {}
+
+        with open(variables_conf_path) as fd:
+            self._parse(variables_conf_path, fd)
+
+    def _parse(self, label, fd):
+
+        for nr, line in enumerate(fd.readlines(), 1):
+
+            line = line.strip()
+
+            if not line or line.startswith('#'):
+                continue
```

Comment:
You mean trailing comments after valid syntax? I don't think that is supported by `ckstat` itself. Also not documented as supported.

It could, however, cause errors with valid syntax, even if not very probably to occur, like:

/my#path/with#hashes root:root 0755

Reply:
oh you're right. I don't think I've ever seen a path like this, but I'll make a mental note, that could actually be helpful for attacking something :)